// Copyright 2010-2025 Google LLC
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package operations_research.sat;

// An arc of a routes constraint, with its LP value.
message ArcLpValue {
  optional int32 tail = 1;
  optional int32 head = 2;
  optional double lp_value = 3;
}

// The arcs of a routes constraint which have non-zero LP values, in the LP
// relaxation of the problem.
message RoutesSupportGraphProto {
  repeated ArcLpValue arc_lp_values = 1;
}
